src/AnalysisPrograms/Recognizers/GenericRecognizer.cs

```diff
@@ -139,35 +139,32 @@ public static void ValidateProfileTagsMatchAlgorithms(Dictionary<string, object>
             var postprocessingConfig = configuration.PostProcessing;
             var postEvents = new List<EventCommon>();
 
-            // count number of events detected at each decibel threshold.
-            for (int i = 1; i <= 39; i++)
-            {
-                var dbEvents = EventFilters.FilterOnDecibelDetectionThreshold(results.NewEvents, (double)i);
+            var groups = results.NewEvents.GroupBy(x => x.DecibelDetectionThreshold);
 
-                if (dbEvents.Count > 0)
-                {
-                    //Log.Debug($"Profiles detected {dbEvents.Count} events at threshold {i} dB.");
-                    var ppEvents = EventPostProcessing.PostProcessingOfSpectralEvents(
-                        dbEvents,
-                        postprocessingConfig,
-                        (double)i,
-                        results.Sonogram,
-                        segmentStartOffset);
-
-                    postEvents.AddRange(ppEvents);
-                }
+            foreach (var group in groups)
+            {
+                var key = group.Key;
+                List<EventCommon> events = group.ToList<EventCommon>();
+                //Log.Debug($"Profiles detected {events.Count} events at threshold {key} dB.");
+
+                var ppEvents = EventPostProcessing.PostProcessingOfSpectralEvents(
+                    events,
+                    postprocessingConfig,
+                    key.Value,
+                    results.Sonogram,
+                    segmentStartOffset);
+
+                postEvents.AddRange(ppEvents);
             }
 
             // Running profiles with multiple dB thresholds produces nested (Russian doll) events.
             // Remove all but the outermost event.
-            // Add a spacer for easier reading of the debug output.
-            Log.Debug($" ");
-            Log.Debug($"Event count BEFORE removing enclosed events = {postEvents.Count}.");
+            Log.Debug($"\nFLATTEN RUSSIAN DOLL EVENTS.");
```

Comment:
Inappropriate terminology.

```suggestion
            Log.Debug($"\nFlatten nested events.");
```

Reply:
Word "flatten" has now also been removed.